src/lib.rs

```diff
@@ -139,7 +142,9 @@ pub trait Watcher: Sized {
     /// Create a new watcher in _immediate_ mode.
     ///
     /// Events will be sent using the provided `tx` immediately after they occur.
-    fn new_immediate(tx: Sender<Result<Event>>) -> Result<Self>;
+    fn new_immediate<F>(event_fn: F) -> Result<Self>
+    where
+        F: EventFn;
```

Comment:
Heh, on one hand, making this generic doesn't make sense, as we use dynamic dipatch internally anyway. On the other hand, generics give us a nicer syntax, where you can pass-in the lambda directly. I don't have a specific advice here, but I'd probably try to arrange API in such a way that publich API is generic, but immediately boxes the closure into an Arc, and calls the private impl

Reply:
Haha yes, I have been conscious of this in each of the backends, that is each of the implementations almost immediately call some function into which the generic function is either `Box`d or `Arc`d depending on the needs of the backend. I figure this way we get the dynamic dispatch indirection to avoid large monomorphisation while still getting that nice generic API and only a single point of dynamic dispatch indirection. You are right about the `PollWatcher` backend though! I just pushed a commit that addresses that backend by having the `PollWatcher::with_delay` function accept an `Arc<dyn EventFn>` instead.

I'll work on addressing some of your other comments soon - thanks for taking a look!